rgdps/repositories/user_relationship.py

```diff
@@ -0,0 +1,234 @@
+    ctx: Context,
+    user_id: int,
+    relationship_type: UserRelationshipType,
+    page: int,
+    page_size: int,
+    include_deleted: bool = False,
+) -> list[UserRelationship]:
+    condition = ""
+    if not include_deleted:
+        condition = "AND deleted = 0"
+
+    relationships_db = await ctx.mysql.fetch_all(
+        "SELECT id, relationship_type, user1_id, user2_id, post_ts, seen_ts "
+        "FROM user_relationships WHERE user1_id = :user1_id AND "
+        f"relationship_type = :relationship_type {condition} "
+        "ORDER BY post_ts DESC LIMIT :limit OFFSET :offset",
+        {
+            "user_id": user_id,
+            "relationship_type": relationship_type.value,
+            "limit": page_size,
+            "offset": page * page_size,
+        },
+    )
+
+    return [
+        UserRelationship.from_mapping(relationship_db)
+        for relationship_db in relationships_db
+    ]
+
+
+async def get_user_relationship_count(
+    ctx: Context,
+    user_id: int,
+    relationship_type: UserRelationshipType,
+    is_new: bool = False,
+    include_deleted: bool = False,
+) -> int:
+
+    condition = ""
+    if not include_deleted:
+        condition = "AND deleted = 0"
+
+    if is_new:
+        condition += " AND seen_ts IS NULL"
+
+    return await ctx.mysql.fetch_val(
+        "SELECT COUNT(*) FROM user_relationships WHERE user1_id = :user_id "
+        f"AND relationship_type = :relationship_type {condition}",
+        {"user_id": user_id, "relationship_type": relationship_type.value},
+    )
+
+
+async def check_relationship_exists(
+    ctx: Context,
+    user1_id: int,
+    user2_id: int,
+    relationship_type: UserRelationshipType,
+    include_deleted: bool = False,
+) -> bool:
+
+    condition = ""
+    if not include_deleted:
+        condition = "AND deleted = 0"
+
+    return await ctx.mysql.fetch_val(
+        "SELECT EXISTS(SELECT 1 FROM user_relationships WHERE user1_id = :user1_id "
+        f"AND user2_id = :user2_id AND relationship_type = :relationship_type {condition})",
+        {
+            "user1_id": user1_id,
+            "user2_id": user2_id,
+            "relationship_type": relationship_type.value,
+        },
+    )
+
+
+async def mark_all_as_seen(
+    ctx: Context,
+    user_id: int,
+    relationship_type: UserRelationshipType,
+    seen_ts: datetime,
+    include_deleted: bool = False,
+) -> None:
+    condition = ""
+    if not include_deleted:
+        condition = "AND deleted = 0"
+
+    await ctx.mysql.execute(
+        "UPDATE user_relationships SET seen_ts = :seen_ts WHERE user1_id = :user_id "
+        f"AND relationship_type = :relationship_type AND seen_ts IS NULL {condition}",
+        {
+            "seen_ts": seen_ts,
+            "user_id": user_id,
+            "relationship_type": relationship_type.value,
+        },
+    )
+
+
+async def create(
+    ctx: Context,
+    user1_id: int,
+    user2_id: int,
+    relationship_type: UserRelationshipType,
```

Comment:
I'd add seen_ts as a possible filed. Might come useful for things such as database conversions.

Reply:
resolved in afd1488